Automattic/jetpack
# CRM: Updating deprecated style handles by passing style handles into wp_print_styles

Fixes https://github.com/Automattic/zero-bs-crm/issues/3368

## Proposed changes:

### The issue:
* On several CRM pages, we use `wp_print_styles` to print out enqueued styles on the page. Testing with WordPress 6.4-RC1, loading each of those pages results in the following PHP error notice in my error log:
 `PHP Deprecated:  Function print_emoji_styles is deprecated since version 6.4.0! Use wp_enqueue_emoji_styles instead. in /var/www/html/wp-includes/functions.php on line 6031`
* This is due to `print_emoji_styles` being deprecated in 6.4, [along with several other functions](https://make.wordpress.org/core/2023/10/17/replacing-hard-coded-style-tags-with-wp_add_inline_style/).

### The puzzle:

* However, we do not hook `print_emoji_styles` into `wp_print_styles` anywhere. Even with all plugins deactivated on my test site including CRM, `print_emoji_styles` is still being hooked into `wp_print_styles`, and it's not coming from the theme either.  If I add `wp_print_styles` to any other page, such as the WP dashboard pages, the same deprecation notice exists.
* In the core WordPress files (6.4-RC1), in `wp-includes/default-filters.php`, the following line exists: `add_action( 'wp_print_styles', 'print_emoji_styles' ); // Retained for backwards-compatibility. Unhooked by wp_enqueue_emoji_styles().`, and `wp_enqueue_emoji_styles` exists on the line above as `add_action( 'wp_enqueue_scripts', 'wp_enqueue_emoji_styles' );` .  `wp_enqueue_emoji_styles` is defined in `formatting.php` and does remove `print_emoji_styles`.

### Solutions:

* One solution was to hard-code the relevant styles into each page and remove the `wp_print_styles` calls. This works.
* Another solution is to add the code suggestion [made on this page](https://make.wordpress.org/core/2023/10/17/replacing-hard-coded-style-tags-with-wp_add_inline_style/) , minus the calls to `ob_start` and `ob_get_clean` which result in broken styling
* **The solution I've chosen** is to pass in the relevant style handles into `wp_print_styles` as an array ([see docs](https://developer.wordpress.org/reference/functions/wp_print_styles/)), which ensures they are the only styles printed. `wp_print_styles` itself is not being deprecated.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/zero-bs-crm/issues/3368
https://github.com/Automattic/jetpack/issues/33655


## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

To test, on a test site running the [WordPress 6.4 release candidate ](https://wordpress.org/news/2023/10/wordpress-6-4-release-candidate-1/)- load the following pages:
* Install and activate the current CRM version for the first time or visit `/wp-admin/admin.php?page=zerobscrm-dash&jpcrm_force_wizard=1` to force the welcome wizard. Once the welcome wizard screen is showing, check your error log - there should be a deprecation notice saying that  `print_emoji_styles` is deprecated.
* Deactivate CRM from the plugins page, and once the 'Before You Go' screen displays check your error log - the deprecation notice should show again.
* Visit the manage forms page - `wp-admin/admin.php?page=manage-forms` and make sure you have one of each of the three types of forms - 'simple', 'naked' and 'content grab'. 
* Visit each form at the relevant link for their type, eg  `/crmforms/content/?fid={formid}`, `/crmforms/naked/?fid={formid}` and `/crmforms/simple/?fid={formid}` an then check your error log - the same deprecation notice should show.
* Test the 'Welcome to Woo Sync' page, though this doesn't show normally at the moment due to commented out code. To test it on a new site setup, un-comment the `require_once` lines in the`show_welcome_wizard` function, in `modules/woo-sync/includes/class-woo-sync.php`
* Test the same pages with this branch checked out, and those deprecation notices should not appear.
